internal/service/federatedsettingsidentityprovider/model_federated_settings_identity_provider.go

```diff
@@ -0,0 +1,285 @@
+				"issuer_uri":                   federatedSettingsIdentityProvider[i].IssuerUri,
+				"okta_idp_id":                  federatedSettingsIdentityProvider[i].OktaIdpId,
+				"pem_file_info":                FlattenPemFileInfo(*federatedSettingsIdentityProvider[i].PemFileInfo),
+				"request_binding":              federatedSettingsIdentityProvider[i].RequestBinding,
+				"response_signature_algorithm": federatedSettingsIdentityProvider[i].ResponseSignatureAlgorithm,
+				"sso_debug_enabled":            federatedSettingsIdentityProvider[i].SsoDebugEnabled,
+				"sso_url":                      federatedSettingsIdentityProvider[i].SsoUrl,
+				"status":                       federatedSettingsIdentityProvider[i].Status,
+			}
+		}
+	}
+
+	return federatedSettingsIdentityProviderMap
+}
+
+func FlattenAssociatedOrgs(associatedOrgs []admin.ConnectedOrgConfig) []map[string]any {
+	var associatedOrgsMap []map[string]any
+
+	if len(associatedOrgs) == 0 {
+		return nil
+	}
+	associatedOrgsMap = make([]map[string]any, len(associatedOrgs))
+
+	for i := range associatedOrgs {
+		if associatedOrgs[i].UserConflicts == nil {
+			associatedOrgsMap[i] = map[string]any{
+				"domain_allow_list":          associatedOrgs[i].DomainAllowList,
+				"domain_restriction_enabled": associatedOrgs[i].DomainRestrictionEnabled,
+				"identity_provider_id":       associatedOrgs[i].IdentityProviderId,
+				"org_id":                     associatedOrgs[i].OrgId,
+				"post_auth_role_grants":      associatedOrgs[i].PostAuthRoleGrants,
+				"role_mappings":              FlattenAuthFederationRoleMapping(associatedOrgs[i].RoleMappings),
+				"user_conflicts":             nil,
+			}
+		} else {
+			associatedOrgsMap[i] = map[string]any{
+				"domain_allow_list":          associatedOrgs[i].DomainAllowList,
+				"domain_restriction_enabled": associatedOrgs[i].DomainRestrictionEnabled,
+				"identity_provider_id":       associatedOrgs[i].IdentityProviderId,
+				"org_id":                     associatedOrgs[i].OrgId,
+				"post_auth_role_grants":      associatedOrgs[i].PostAuthRoleGrants,
+				"role_mappings":              FlattenAuthFederationRoleMapping(associatedOrgs[i].RoleMappings),
+				"user_conflicts":             FlattenFederatedUser(associatedOrgs[i].UserConflicts),
+			}
+		}
+	}
+
+	return associatedOrgsMap
+}
+
+type mRoleAssignmentV2 []admin.RoleAssignment
+
+func (ra mRoleAssignmentV2) Len() int      { return len(ra) }
+func (ra mRoleAssignmentV2) Swap(i, j int) { ra[i], ra[j] = ra[j], ra[i] }
+func (ra mRoleAssignmentV2) Less(i, j int) bool {
+	compareVal := strings.Compare(ra[i].GetOrgId(), ra[j].GetOrgId())
+
+	if compareVal != 0 {
+		return compareVal < 0
+	}
+
+	compareVal = strings.Compare(ra[i].GetGroupId(), ra[j].GetGroupId())
+
+	if compareVal != 0 {
+		return compareVal < 0
+	}
+
+	return *ra[i].Role < *ra[j].Role
+}
+
+func FlattenRoleAssignmentsV2(roleAssignments []admin.RoleAssignment) []map[string]any {
+	sort.Sort(mRoleAssignmentV2(roleAssignments))
```

Comment:
I think this was just moved.   It's not bad to implement the sort interface, but alternatively it can be done in this way and you only need to implement the less functionality: 
```
	sort.Slice(mRoleAssignmentV2, func(i, j int) bool {
// less logic ...
}
```

Reply:
Yes this was moved,. I kept it like this to avoid changing more things, but good to know there's a way to do it implementing less funcitonality